### PR TITLE
Clean dictionaries Fastfile

### PR DESCRIPTION
**Pull request by pablog**
Same idea as before.

Removed dictionaries from `Fastfile` and created custom actions instead.

Next step: create `Fastfile.private` and keep `Fastfile` for public lanes.

### CODE REVIEW

File: Fastfile [hunk 1]
```diff
@@ -268,7 +246,7 @@ reflected in `Info.plist` during building."
 
     pem(
       generate_p12: true,
-      development: match_types[build_configuration] == "development",
+      development: get_match_type(build_configuration: build_configuration) == "development",
```
Comment (danielaRiesgo) [inline Fastfile]: This "development" should be replaced by `GetMatchTypeAction .MATCH_TYPES[:test]`, right?

**Review by danielaRiesgo — COMMENTED** [review]
The only thing I don't like about this, is that all the dictionaries are dependent: they must match keys  to build configurations and that. So if we want to change a name or something about one dictionary, we may forget to go update the others since there is nothng indicating you that relationship. Specially if you are a new developer doing the kickoff with wolox's tools you don't know yet.

File: Fastfile (hunk 1 above)
Comment (pablog) [reply Fastfile]: `development` is a `Bool` type. This comparison is right.

File: Fastfile [hunk 2]
```diff
@@ -81,7 +58,8 @@ platform :ios do
   end
 
   desc "Releases a new version to `TestFlight`. This lane must receive the following parameters:"
-  desc "- build_configuration: A build configuration to deploy. Can be any of: `%s`" % build_configurations.keys.to_s
+  desc "- build_configuration: A build configuration to deploy. \
+Can be any of: `%s`" % Actions::GetBuildConfigurationAction::BUILD_CONFIGURATIONS.keys.to_s
```
Comment (dtilve) [inline Fastfile]: I feel there's a space or something missing here

**Review by dtilve — APPROVED** [review]
👍  Loving this clean up

File: Fastfile (hunk 1 above)
Comment (dtilve) [reply Fastfile]: I like it too 🎉 👍

File: Fastfile (hunk 2 above)
Comment (pablog) [reply Fastfile]: No, this is the right way to do it. I don't like either the way it looks in the code, but the documentation is properly generated.

**Comment by pablog** [comment]
@danielaRiesgo:

```
The only thing I don't like about this, is that all the dictionaries are dependent: they must match keys to build configurations and that. So if we want to change a name or something about one dictionary, we may forget to go update the others since there is nothng indicating you that relationship. Specially if you are a new developer doing the kickoff with wolox's tools you don't know yet.
```

I fully agree with this. This is a problem we already had before this refactor, and that always made me think there was something not properly modeled. Anyway, those keys are super coupled to everything, in particular xcode project configurations. But, if possible I'd like to have them declared in a single dictionary.

A possible proposal, not sure if worth the effort: 

- Declare a single dictionary that links xcode configurations -> fastlane internal configurations.
- Make each of these dictionaries use these fastlane internal configurations. This way, if we need to change the configurations from xcode (hope never happens again) we only will need to change one dictionary.

Can you suggest anything better or a different idea? I'd like to refactor that too.

**Comment by danielaRiesgo** [comment]
@pablog yeah, the thing is that now, all dictionaries are separated so it's even easier to forget to change something. Ideally, if it was something like Swift, I would make the enum with an extension for each thing needed: build_configuration, match_type, project_name, etc. All in one file. I'm not sure how to transform that in ruby.

**Review by damfinkel — APPROVED** [review]
nice!

**Comment by pablog** [comment]
@danielaRiesgo I've been thinking about this, and don't think there's a proper way in Ruby to do this.

In my opinion we should write tests in rspec to ensure this is working.

**Review by danielaRiesgo — APPROVED** [review]

**Comment by pablog** [comment]
I created an issue for testing: https://github.com/Wolox/fastlane-mobile/issues/8

You're all invited to give it a try.

**Pull request merged**